### ISSUE TITLE
Change Slides To Show Dynamically

### ISSUE THREAD

**Issue by gweiss27** [vasyabigi/angular-slick#135]
Hi there,

Love your carousel!  

Question:

I would like to change the Sides to Show dynamically as a user resizes the viewport.  So that 8 slides show on widescreens, but 6 or less show when making the window smaller.  I created a scope variable for the slide to show and can set it initially on the inital viewport size.  I also am detecting the resize event and the window size, but I'm not able to get the slick directive to actually change the slides to show when I make these changes.

Do you have any idea how this might be done?

Thanks!

Greg

**Comment by jkbyln** [comment]
Figured this out after struggling with it for a few days~

Turns out Slick.js has a setting called 'responsive' that's already supported by this plugin. You can pass in settings to change at different breakpoints.

Can be used like so:

```
<slick dots="false"
    slides-to-show="5"
    slides-to-scroll="5"
    ng-show="loaded"
    data="AdProducts"
    responsive="[{
        breakpoint: 1024,
        settings: { slidesToShow: 3, slidesToScroll: 3, }
    }, {
        breakpoint: 600,
        settings: { slidesToShow: 1, slidesToScroll: 1 }
    }]">
```

**Comment by gweiss27** [comment]
Yes, I figured it out to after much research.  Would be nice to have this front and center in their docs.  But I got it to work as well.  Thanks for the response!

**Comment by alexswiontek** [comment]
I've found that using dynamic data causes a flash of angular logic before the slider starts, and that using [ngCloak](https://docs.angularjs.org/api/ng/directive/ngCloak) will fix that. Like so:

``` js
<slick ng-cloak></slick>
```

Not sure if this was mentioned elsewhere, but it helped me out—hopefully it can help someone else out as well. Good to note that in order for this to work, the script for Angular must be loaded in the `<head>`.
